exchanges/okx/okx_test.go

```diff
@@ -3413,250 +4735,1341 @@ func TestGetAssetsFromInstrumentTypeOrID(t *testing.T) {
-	if err != nil {
-		t.Error(err)
-	}
+	require.ErrorIs(t, err, asset.ErrNotSupported)
 
-	_, err = ok.GetFuturesPositionOrders(contextGenerate(), &futures.PositionsRequest{
-		Asset:     asset.Spot,
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, ok)
+	result, err := ok.GetFuturesPositionOrders(contextGenerate(), &futures.PositionsRequest{
+		Asset:     asset.Futures,
 		Pairs:     []currency.Pair{pp[0]},
 		StartDate: time.Now().Add(time.Hour * 24 * -7),
+		EndDate:   time.Now(),
 	})
-	if !errors.Is(err, futures.ErrNotFuturesAsset) {
-		t.Errorf("received '%v', expected '%v'", err, asset.ErrNotSupported)
-	}
+	require.NoError(t, err)
+	assert.NotNil(t, result)
 }
 
 func TestGetLeverage(t *testing.T) {
 	t.Parallel()
+	pp, err := ok.CurrencyPairs.GetPairs(asset.Futures, true)
+	require.NoError(t, err)
+	_, err = ok.GetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, order.UnknownSide)
+	require.ErrorIs(t, err, order.ErrSideIsInvalid)
+
 	sharedtestvalues.SkipTestIfCredentialsUnset(t, ok)
+	result, err := ok.GetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Multi, order.UnknownSide)
+	require.NoError(t, err)
+	require.NotNil(t, result)
+	result, err = ok.GetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, order.Long)
+	require.NoError(t, err)
+	require.NotNil(t, result)
+
+	result, err = ok.GetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, order.Short)
+	require.NoError(t, err)
+	assert.NotNil(t, result)
+}
+
+func TestSetLeverage(t *testing.T) {
+	t.Parallel()
 	pp, err := ok.CurrencyPairs.GetPairs(asset.Futures, true)
-	if err != nil {
-		t.Error(err)
+	require.NoError(t, err)
+	err = ok.SetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, 5, order.UnknownSide)
+	require.ErrorIs(t, err, order.ErrSideIsInvalid)
+	err = ok.SetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, 5, order.CouldNotBuy)
+	require.ErrorIs(t, err, order.ErrSideIsInvalid)
+	err = ok.SetLeverage(contextGenerate(), asset.Spot, pp[0], margin.Multi, 5, order.UnknownSide)
+	require.ErrorIs(t, err, asset.ErrNotSupported)
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, ok, canManipulateRealOrders)
+	err = ok.SetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Multi, 5, order.UnknownSide)
+	require.NoError(t, err)
+	err = ok.SetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, 5, order.Long)
+	require.NoError(t, err)
+	err = ok.SetLeverage(contextGenerate(), asset.Futures, pp[0], margin.Isolated, 5, order.Short)
+	assert.NoError(t, err)
+}
+
+func TestGetFuturesContractDetails(t *testing.T) {
+	t.Parallel()
+	_, err := ok.GetFuturesContractDetails(context.Background(), asset.Spot)
+	require.ErrorIs(t, err, futures.ErrNotFuturesAsset)
+	_, err = ok.GetFuturesContractDetails(context.Background(), asset.USDTMarginedFutures)
+	require.ErrorIs(t, err, asset.ErrNotSupported)
+
+	result, err := ok.GetFuturesContractDetails(context.Background(), asset.Futures)
```

Comment:
Can be simplified to:

```suggestion
	func TestGetFuturesContractDetails(t *testing.T) {
	t.Parallel()
	_, err := ok.GetFuturesContractDetails(context.Background(), asset.Spot)
	require.ErrorIs(t, err, futures.ErrNotFuturesAsset)
	_, err = ok.GetFuturesContractDetails(context.Background(), asset.USDTMarginedFutures)
	require.ErrorIs(t, err, asset.ErrNotSupported)

	for _, a := range []asset.Item{asset.Futures, asset.PerpetualSwap, asset.Spread} {
		result, err := ok.GetFuturesContractDetails(context.Background(), a)
		require.NoError(t, err)
		require.NotNil(t, result)
	}
}
```